index.js

```diff
@@ -200,10 +201,20 @@ export default function normalizeUrl(urlString, options) {
 		}
 	}
 
-	if (options.removeQueryParameters === true) {
+	if (options.keepQueryParameters.length === 0 && options.removeQueryParameters === true) {
```

Comment:
I think it should be a `Array.isArray` check. We still want it to take precedence, even if it's empty.

Reply:
The default value of `keepQueryParameters` is `[]`.
`Array.isArray(options.keepQueryParameters)` returns `true` even the array is empty.

```javascript
Array.isArray([]) // true
```

Moving the `length` check from the `Array.isArray` check still makes it to be entered in the if block.
```diff
- if (options.keepQueryParameters.length === 0 && options.removeQueryParameters === true) {
+ if (!Array.isArray(options.keepQueryParameters) && options.removeQueryParameters === true) {
    urlObject.search = '';
}
```

What do you suggest?

Reply:
Yeah, I'm suggesting that the internal default would be `undefined` (meaning, just remove line 69).

My reasoning is that if the user specifies an empty array, they mean that no query parameters should be kept. Does that make sense?

Reply:
Yes, it makes sense. 👍 
I made the changes (109567a281d3d09ff50d7aa4e63357df2eccc603)